src/cloudformation/structure/cloudformation_parser_test.go

```diff
@@ -85,64 +109,101 @@ func Test_mapResourcesLineYAML(t *testing.T) {
+	cfnParser := CloudformationParser{}
+	cfnParser.Init(directory, nil)
+	readFilePath := directory + "/" + testFileName + "." + fileType
+	tagGroup := simple.TagGroup{}
+	extraTags := []tags.ITag{
+		&tags.Tag{
+			Key:   "new_tag",
+			Value: "new_value",
+		},
+	}
+	tagGroup.SetTags(extraTags)
+	tagGroup.InitTagGroup("", []string{})
+	writeFilePath := directory + "/" + testFileName + "_tagged." + fileType
+	cfnBlocks, err := cfnParser.ParseFile(readFilePath)
+	for _, block := range cfnBlocks {
+		err := tagGroup.CreateTagsForBlock(block)
 		if err != nil {
 			t.Fail()
 		}
-		defer func() {
-			_ = expectedHandler.Close()
-			_ = actualHandler.Close()
-			_ = os.Remove(f.Name())
-		}()
-		actualReader := bufio.NewScanner(actualHandler)
-		expectedReader := bufio.NewScanner(expectedHandler)
-		for actualReader.Scan() && expectedReader.Scan() {
-			actualLine := actualReader.Text()
-			expectedLine := expectedReader.Text()
-			assert.Equal(t, strings.Trim(actualLine, " \n\t"), strings.Trim(expectedLine, " \n\t"))
-		}
+	}
+	if err != nil {
+		t.Fail()
+	}
+	_, err = f.Seek(0, io.SeekStart)
+	if err != nil {
+		t.Fail()
+	}
+	err = cfnParser.WriteFile(readFilePath, cfnBlocks, f.Name())
+	if err != nil {
+		t.Fail()
+	}
+	var expectedHandler, actualHandler *os.File
+	expectedAbs, _ := filepath.Abs(writeFilePath)
+	actualAbs, _ := filepath.Abs(f.Name())
+	expectedHandler, _ = os.OpenFile(expectedAbs, os.O_RDWR, 0755)
+	actualHandler, _ = os.OpenFile(actualAbs, os.O_RDWR|os.O_CREATE, 0755)
+	_, err = expectedHandler.Seek(0, io.SeekStart)
+	if err != nil {
+		t.Fail()
+	}
+	_, err = actualHandler.Seek(0, io.SeekStart)
+	if err != nil {
+		t.Fail()
+	}
+	defer func() {
+		_ = expectedHandler.Close()
+		_ = actualHandler.Close()
+		_ = os.Remove(f.Name())
+	}()
+	actualReader := bufio.NewScanner(actualHandler)
+	expectedReader := bufio.NewScanner(expectedHandler)
+	for actualReader.Scan() && expectedReader.Scan() {
+		actualLine := actualReader.Text()
+		expectedLine := expectedReader.Text()
+		assert.Equal(t, strings.Trim(actualLine, " \n\t"), strings.Trim(expectedLine, " \n\t"))
+	}
+}
+
+func TestWriteCFN(t *testing.T) {
+
+	t.Run("test CFN yaml writing", func(t *testing.T) {
+		directory := "../../../tests/cloudformation/resources/ebs"
+
+		writeCFNTestHelper(t, directory, "ebs", "yaml")
 	})
+
+	t.Run("test pre-tagged CFN json writing", func(t *testing.T) {
+		directory := "../../../tests/cloudformation/resources/ebs"
+		writeCFNTestHelper(t, directory, "ebs", "json")
+	})
+
+	t.Run("test untagged CFN json writing", func(t *testing.T) {
+		directory := "../../../tests/cloudformation/resources/ec2_untagged"
+		writeCFNTestHelper(t, directory, "ec2_untagged", "json")
+	})
+
+	t.Run("test untagged CFN json writing with bad indentation", func(t *testing.T) {
+		directory := "../../../tests/cloudformation/resources/json"
+		writeCFNTestHelper(t, directory, "SingleENIwithMultipleEIPs", "json")
+	})
```

Comment:
@nimrodkor I don't think so because there's a yaml test as well. I just used an existing yaml test to create infrastructure for any cfn file type test

Reply:
@rotemavni I understand, however it seems the coverage in the json package is a bit on the low side:
```
ok      github.com/bridgecrewio/yor/src/common/json     1.270s  coverage: 14.9% of statements
```

Reply:
@nimrodkor How is this?
![image](https://user-images.githubusercontent.com/52502521/120314625-1bd1d300-c2e4-11eb-9732-0af56dfd76f9.png)